app/controllers/api/v1/offers_controller.rb

```diff
@@ -40,6 +40,9 @@ class OffersController < Api::V1::ApiController
       param_group :offer
       def create
         @offer.created_by_id = current_user.id unless offer_params.has_key?(:created_by_id)
+        if @offer.created_by_id.blank? && @offer.under_review?
```

Comment:
This fix works.

Putting out thoughts related to the existing implementation:

We first create an empty offer, and then as we add details, the offer is updated.  (Which I feel is wrong)
Ideally it should not even create an offer record if everything is empty. Only after an item is added or any other details are added, only then it should create it.

Should we change our current implementation over here? 

FYI, everything above can be thought of a separate ticket.

cc: @steveyken @patrixr @abulsayyad123 @shreyas098 @swatijadhav

Reply:
Good point to raise. Is there a reason that we allow admins to create blank offers? I know we do for donors to smooth their experience but it is necessary for admins also?

Reply:
I think we definitely want to avoid something like this. Generally any controller code that is written `if (some very unique state)` is dangerous.

I think the potential fixes are:
- Create a non-empty offer from the start (as @bharat619 suggests)
- Prompt the admin to select his/her existing offers when he/she tries to create another offer

More generally:
An admin **is allowed** to create an empty offer, so the behaviour of `POST /offers` should not change.
We can't have an endpoint that _pretends_ to create something but actually returns something else.
That would mean the server is making an _assumption_ on what the user is trying to do, and that's a poor api design choice (and a ticking time bomb)

Do make sure to get @steveyken 's input on this. Matt can also help to figure out a nice UI flow to avoid that problem.